integration-test/src/test/java/org/apache/iotdb/pipe/it/IoTDBPipeAlterIT.java

```diff
@@ -103,16 +104,62 @@ public void testBasicPipeAlter() throws Exception {
       List<TShowPipeInfo> showPipeResult = client.showPipe(new TShowPipeReq()).pipeInfoList;
       Assert.assertEquals(1, showPipeResult.size());
       Assert.assertTrue(showPipeResult.get(0).pipeConnector.contains("batch.enable=true"));
+      // alter pipe (modify) will keep unspecified configurations
+      Assert.assertTrue(
+          showPipeResult
+              .get(0)
+              .pipeConnector
+              .contains(String.format("node-urls=%s", receiverDataNode.getIpAndPortString())));
+      // alter pipe will keep user stopped status
       Assert.assertEquals("STOPPED", showPipeResult.get(0).state);
       // alter pipe will reset pipe creation time
-      Assert.assertTrue(showPipeResult.get(0).creationTime > firstCreationTime);
+      Assert.assertTrue(showPipeResult.get(0).creationTime > lastCreationTime);
+      lastCreationTime = showPipeResult.get(0).creationTime;
+      // alter pipe will clear exception messages
+      Assert.assertEquals("", showPipeResult.get(0).exceptionMessage);
+    }
+
+    // start pipe
+    try (Connection connection = senderEnv.getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute("start pipe a2b");
+    } catch (SQLException e) {
+      fail(e.getMessage());
+    }
+
+    // alter pipe (replace)
+    try (Connection connection = senderEnv.getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute("alter pipe a2b replace processor ('processor'='down-sampling-processor')");
+    } catch (SQLException e) {
+      fail(e.getMessage());
+    }
+
+    // show pipe
+    try (SyncConfigNodeIServiceClient client =
+        (SyncConfigNodeIServiceClient) senderEnv.getLeaderConfigNodeConnection()) {
+      List<TShowPipeInfo> showPipeResult = client.showPipe(new TShowPipeReq()).pipeInfoList;
+      Assert.assertEquals(1, showPipeResult.size());
+      Assert.assertTrue(
+          showPipeResult.get(0).pipeProcessor.contains("processor=down-sampling-processor"));
+      // alter pipe (without sink clause) will not modify sink's configurations
+      Assert.assertTrue(showPipeResult.get(0).pipeConnector.contains("batch.enable=true"));
+      Assert.assertTrue(
+          showPipeResult
+              .get(0)
+              .pipeConnector
+              .contains(String.format("node-urls=%s", receiverDataNode.getIpAndPortString())));
+      // alter pipe will keep running status
+      Assert.assertEquals("RUNNING", showPipeResult.get(0).state);
+      // alter pipe will reset pipe creation time
+      Assert.assertTrue(showPipeResult.get(0).creationTime > lastCreationTime);
       // alter pipe will clear exception messages
       Assert.assertEquals("", showPipeResult.get(0).exceptionMessage);
```

Comment:
The exception message may always be empty, whether or not the "alter pipe" clear it.